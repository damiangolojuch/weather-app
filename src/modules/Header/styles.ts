import styled from "styled-components";
import { headerHeight } from "~/components/Layout/styles";
import config from "~/styles/config";
import Logo from "~/components/Logo";
import CitySearchInput from "~/modules/CitySearchInput";
import { Link } from "react-router-dom";

export const LinkStyled = styled(Link)`
  display: inline-flex;
  align-content: flex-start;
`;

export const HeaderWrapper = styled.div`
  height: ${headerHeight}px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 10px 0;
`;

const miniLogoBreakpoint = config.viewports.md;

export const LogoStyled = styled(Logo)`
  height: ${headerHeight - 20}px;
  padding-right: 50px;

  @media (max-width: ${miniLogoBreakpoint - 1}px) {
    display: none;
  }
`;

export const LogoMiniStyled = styled(Logo)`
  height: ${headerHeight - 20}px;
  padding-right: 10px;

  @media (min-width: ${miniLogoBreakpoint}px) {
    display: none;
  }
`;

export const CitySearchInputStyled = styled(CitySearchInput)``;
