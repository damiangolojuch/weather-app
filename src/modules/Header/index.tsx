import { LogoType } from "~/components/Logo";
import * as S from "./styles";

const Header = () => (
  <S.HeaderWrapper>
    <S.LinkStyled to={"/"}>
      <S.LogoMiniStyled logoType={LogoType.mini} />
      <S.LogoStyled />
    </S.LinkStyled>
    <S.CitySearchInputStyled />
  </S.HeaderWrapper>
);

export default Header;
