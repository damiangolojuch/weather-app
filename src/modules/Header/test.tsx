import Header from "~/modules/Header/index";
import { render } from "@testing-library/react";

jest.mock("react-router-dom", () => {
  return {
    __esModule: true,
    Link: (props: any) => <div>{props.children}</div>,
  };
});

jest.mock("modules/CitySearchInput", () => {
  return {
    __esModule: true,
    default: (props: any) => (
      <div data-component="modules/CitySearchInput">{props.children}</div>
    ),
  };
});

describe("<Header />", () => {
  it("should render header with logo and input", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
