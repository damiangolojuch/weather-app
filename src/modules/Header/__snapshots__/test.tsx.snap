// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render header with logo and input 1`] = `
<div>
  <div
    class="sc-kDDrLX bIOnTK"
  >
    <div>
      <img
        class="sc-bczRLJ sc-crXcEl kUGlum"
        src="/assets/logo-mini.svg"
      />
      <img
        class="sc-bczRLJ sc-iqcoie dcMPdw"
        src="/assets/logo.svg"
      />
    </div>
    <div
      data-component="modules/CitySearchInput"
    />
  </div>
</div>
`;
